test/integration/cds_integration_test.cc

```diff
@@ -195,7 +195,7 @@ TEST_P(CdsIntegrationTest, CdsClusterUpDownUp) {
 TEST_P(CdsIntegrationTest, CdsClusterTeardownWhileConnecting) {
   initialize();
   test_server_->waitForCounterGe("cluster_manager.cluster_added", 1);
-  test_server_->waitForCounterExists("cluster.cluster_1.upstream_cx_total");
+  EXPECT_NE(test_server_->counter("cluster.cluster_1.upstream_cx_total"), nullptr);
```

Comment:
this changes the function of this test in a way we might not like. Current behavior is to block at line 198 until another thread creates the stat. There might be a race this resolves.

You just fast fail if the thread creating the stat gets ahead of the test thread.